packages/core/src/indexing/database-index-api.ts

```diff
@@ -59,7 +81,7 @@ export abstract class DatabaseIndexApi<DateType = Date | number> {
   private modelsToIndex: Array<StreamID> = []
   // Maps Model streamIDs to the list of fields in the content of MIDs in that model that should be
   // indexed
-  private readonly modelsIndexedFields = new Map<string, Array<string>>()
+  private readonly modelsIndexedFields = new Map<StreamID, Array<string>>()
```

Comment:
Does the mapping work as expected for different `StreamID` instances?

Reply:
Yeah, I'm pretty sure if you pass two different StreamID instances that represent the same actual StreamID, they'll go in as different fields in the Map, since I believe by default Objects use pointer memory location equality when they are keys in a map unless you specify a custom comparison function.  @ukstv would know for sure or you could do a simple test to confirm